### PR TITLE
Allow newer versions of simsimd (<7.x)

### LINKED ISSUE #212: Cannot install on Python 3.13
**Describe the bug**
The version constraint of `simsimd<5.0.0` prevents installing `langchain-weaviate` on Python >= 3.13, due to a bug in `simsimd` in `v4.4.0` (and earlier).

**Complete Minimal Reproducible Example**
1. Try to install `langchain-weaviate` on any platform for which no wheels of `simsimd==4.4.0` are available (e.g. Python 3.13, but also older or more rare setups).
2. Get an error like this:

```
#13 0.112 Using CPython 3.13.0 interpreter at: /usr/local/bin/python3
#13 0.112 Creating virtual environment at: .venv
#13 0.861   × Failed to download and build `simsimd==4.4.0`
#13 0.861   ╰─▶ Build backend failed to determine requirements with `build_wheel()`
#13 0.861       (exit status: 1)
#13 0.861 
#13 0.861       [stderr]
#13 0.861       Traceback (most recent call last):
#13 0.861         File "<string>", line 14, in <module>
#13 0.861           requires = get_requires_for_build({})
#13 0.861         File
#13 0.861       "/root/.cache/uv/builds-v0/.tmpKjfYuY/lib/python3.13/site-packages/setuptools/build_meta.py",
#13 0.861       line 334, in get_requires_for_build_wheel
#13 0.861           return self._get_build_requires(config_settings, requirements=[])
#13 0.861                  ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
#13 0.861         File
#13 0.861       "/root/.cache/uv/builds-v0/.tmpKjfYuY/lib/python3.13/site-packages/setuptools/build_meta.py",
#13 0.861       line 304, in _get_build_requires
#13 0.861           self.run_setup()
#13 0.861           ~~~~~~~~~~~~~~^^
#13 0.861         File
#13 0.861       "/root/.cache/uv/builds-v0/.tmpKjfYuY/lib/python3.13/site-packages/setuptools/build_meta.py",
#13 0.861       line 320, in run_setup
#13 0.861           exec(code, locals())
#13 0.861           ~~~~^^^^^^^^^^^^^^^^
#13 0.861         File "<string>", line 6, in <module>
#13 0.861           sys.path = [] + sys.path
#13 0.861                         ^^^^^^^^^^
#13 0.861       FileNotFoundError: [Errno 2] No such file or directory: 'VERSION'
#13 0.861 
#13 0.861   help: `simsimd` was included because `example-project==1.0.0` depends on
#13 0.861         `langchain-weaviate==0.0.3` which depends on `simsimd`
```

**Expected behavior**
`langchain-weaviate` should be installable also in environments for which no pre-built `simsimd` wheels are available.

**Tech Stack:**
- weaviate server version: irrelevant
- weaviate client version: `v4.9.3`
- weaviate modules enabled: none
- langchain-weaviate version: `v0.03`
- langchain version: `v0.3.7`

**Additional context**
This bug was reported on the `simsimd` side in https://github.com/ashvardanian/SimSIMD/issues/150 and fixed in `v5.0.0` (and then appeared again in https://github.com/ashvardanian/SimSIMD/issues/209 and fixed again in `v5.7.1`). The current version of `simsimd` is `v6.0.5`.

### PR DESCRIPTION
**Pull request by graipher**
Fixes #212

Increase allowed version range for simsimd to allow for the current 6.x version. This should allow installing this package with Python 3.13.

### CODE REVIEW

**Comment by graipher** [comment]
Arguably, the python version constraint should also be changed to `python = ">=3.9,<3.14"` to make it more explicit that any newer Python version is not automatically usable.

File: libs/weaviate/pyproject.toml [hunk 1]
```diff
@@ -74,7 +74,7 @@ lint.select = [
 ]
 
 [tool.mypy]
-disallow_untyped_defs = "True"
+disallow_untyped_defs = true
```
Comment (graipher) [inline libs/weaviate/pyproject.toml:77]: My editor flagged this, not sure if it is needed.

**Comment by tjmcdonough** [comment]
Can this be merged?

**Comment by graipher** [comment]
In my opinion yes, once the pipeline has succeeded after having been approved to  run.

**Comment by hsm207** [comment]
@graipher thank you for your contribution! This has been taken care in PR #214 so I'll go ahead and close this one.


Sorry I missed your PR earlier. I was cleaning up the codebase and missed this.

Appreciate your effort though!

**Pull request closed without merging**